sklearn/linear_model/tests/test_logistic.py

```diff
@@ -540,6 +540,50 @@ def test_logistic_regressioncv_class_weights():
     assert_array_almost_equal(clf_lib.coef_, clf_lbf.coef_, decimal=4)
 
 
+def _compute_class_weight_dictionary(y):
+    # compute class_weight and return it as a dictionary
+    classes = np.unique(y)
+    class_weight = compute_class_weight("balanced", classes, y)
+
+    class_weight_dict = {}
+    for (cw, cl) in zip(class_weight, classes):
+        class_weight_dict[cl] = cw
+
+    return class_weight_dict
+
+
+def test_logistic_regression_class_weights():
+    # Multinomial case: remove 90% of class 0
+    X = iris.data[45:, :]
+    y = iris.target[45:]
+    solvers = ("lbfgs", "newton-cg")
+    class_weight_dict = _compute_class_weight_dictionary(y)
+
+    for solver in solvers:
+        clf1 = LogisticRegression(solver=solver, multi_class="multinomial",
+                                  class_weight="balanced")
+        clf2 = LogisticRegression(solver=solver, multi_class="multinomial",
+                                  class_weight=class_weight_dict)
+        clf1.fit(X, y)
+        clf2.fit(X, y)
+        assert_array_almost_equal(clf1.coef_, clf2.coef_, decimal=6)
+
+    # Binary case: remove 90% of class 0 and 100% of class 2
```

Comment:
I think I understand the equivalent `class_weight_dict` for 3 classes with OvR is no longer trivial to compute in that case.